### PR TITLE
sql: add `flashback cluster` doc

### PR DESCRIPTION
**Pull request by Defined2014**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/pingcap/tidb/issues/37197

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by shichun-0415** [comment]
Hi @Defined2014. Since this feature is expected to be released in v6.4, do we have an FD jira issue for it? I didn't find one in the v6.4 FD list.

**Comment by Defined2014** [comment]
> Hi @Defined2014. Since this feature is expected to be released in v6.4, do we have an FD jira issue for it? I didn't find one in the v6.4 FD list.

@shichun-0415 The FD number is 943.

File: sql-statements/sql-statement-flashback-cluster-to-timestamp.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+Empty set (0.00 sec)
+```
+
+## 工作原理
+
+在 TiKV 中为了避免数据并发更新时引入锁的开销从而引入了多版本并发控制 (MVCC) 机制。在该机制下，TiDB 对数据进行修改操作时并不会在原始数据库上直接操作，而是会写入一个更新版本的数据将老的数据覆盖。TiDB 后台的 GC Worker 会定期推进 `tikv_gc_safe_point`，并在后台将旧于该时间点的版本删除。`FLASHBACK CLUSTER` 正是基于 TiKV 这一特性实现的。
+
+`FLASHBACK CLUSTER` 可以简单的分为下面几个阶段，
+
+* 在正式开始之前，TiDB 会将 `FLASHBACK` 过程中需要修改的变量值保存在 `Job.Args` 上，并在结束后恢复。
```
Comment (bb7133) [inline sql-statements/sql-statement-flashback-cluster-to-timestamp.md:81]: I'm not sure if putting those internal processes here is helpful. Instead, it's better to describe those details in design doc(RFC) IMO.

File: sql-statements/sql-statement-flashback-cluster-to-timestamp.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+* `FLASHBACK` 指定的时间点需要在 Garbage Collection (GC) life time 时间内，可以使用系统变量 `tidb_gc_life_time` 配置数据的历史版本的保留时间（默认值是 10m0s）。可以使用以下 SQL 语句查询当前的 safePoint，即 GC 已经清理到的时间点：
+
+```sql
+SELECT * FROM mysql.tidb WHERE variable_name = 'tikv_gc_safe_point';
+```
+
+* 仅支持拥有 `SUPER` 权限的用户执行 `FLASHBACK CLUSTER` 操作。
+* 在 `FLASHBACK CLUSTER` 指定的时间点到开始执行的时间段内不能存在非 `FLASHBACK CLUSTER` 类型的 DDL 记录。
+* 在执行 `FLASHBACK CLUSTER` 期间，TiDB 会主动断开所有非 `FLASHBACK CLUSTER` 的链接，并在禁止对相关字段进行读写操作，直到 `FLASHBACK CLUSTER` 完成。
+* `FLASHBACK CLUSTER` 命令不能取消，一旦开始执行会一直重试，直到成功。
```
Comment (bb7133) [inline sql-statements/sql-statement-flashback-cluster-to-timestamp.md:36]: Could we try to add some descriptions on the observability of the process, for example, how to check the progress of flashback even this is not implemented yet?

File: sql-statements/sql-statement-flashback-cluster-to-timestamp.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
++------+
+1 row in set (0.01 sec)
+
+mysql> flashback cluster to timestamp '2022-09-28 17:24:16';
+Query OK, 0 rows affected (0.20 sec)
+
+mysql> select * from t;
+Empty set (0.00 sec)
+```
+
```
Comment (bb7133) [inline sql-statements/sql-statement-flashback-cluster-to-timestamp.md:74]: Could you add an example of 'error case'?

File: sql-statements/sql-statement-flashback-cluster-to-timestamp.md (hunk 1 above)
Comment (Defined2014) [reply sql-statements/sql-statement-flashback-cluster-to-timestamp.md:81]: Just ref https://docs.pingcap.com/zh/tidb/stable/sql-statement-flashback-table#%E5%B7%A5%E4%BD%9C%E5%8E%9F%E7%90%86.
So maybe we could delete those part?

File: sql-statements/sql-statement-flashback-database-to-timestamp.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+```
+
+* 执行 `FLASHBACK DATABASE` 需要有 DATABASE 权限。
+* 在 `FLASHBACK DATABASE` 指定的时间点到开始执行的时间段内不能存在相关的 DDL 执行记录。
+* 在执行 `FLASHBACK DATABASE` 期间，TiDB 会主动断开所有相关的链接和正在执行的事务，直到 `FLASHBACK DATABASE` 完成。
+* `FLASHBACK DATABASE` 命令不能取消，一旦开始执行会一直重试，直到成功。
+
+## 示例
+
+* 恢复新插入的数据
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-flashback-database-to-timestamp.md:40]:
```suggestion
恢复新插入的数据：
```

File: sql-statements/sql-statement-flashback-database-to-timestamp.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+|    1 |
++------+
+1 row in set (0.01 sec)
+
+mysql> flashback database test to timestamp '2022-09-28 17:24:16';
+Query OK, 0 rows affected (0.20 sec)
+
+mysql> select * from t;
+Empty set (0.00 sec)
+```
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-flashback-database-to-timestamp.md:76]:
````suggestion
```sql
mysql> USE test;
Query OK, 0 rows affected (0.00 sec)
mysql> CREATE TABLE t(a INT);
Query OK, 0 rows affected (0.09 sec)
mysql> SELECT * FROM t;
Empty set (0.01 sec)
mysql> SELECT NOW();
+---------------------+
| now()               |
+---------------------+
| 2022-09-28 17:24:16 |
+---------------------+
1 row in set (0.02 sec)
mysql> INSERT INTO t VALUES (1);
Query OK, 1 row affected (0.02 sec)
mysql> SELECT * FROM t;
+------+
| a    |
+------+
|    1 |
+------+
1 row in set (0.01 sec)
mysql> FLASHBACK DATABASE TEST TO TIMESTAMP '2022-09-28 17:24:16';
Query OK, 0 rows affected (0.20 sec)
mysql> SELECT * FROM t;
Empty set (0.00 sec)
```
````

File: sql-statements/sql-statement-flashback-table-to-timestamp.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    "FLASHBACK" "TABLE" TableNameList "TO" "TIMESTAMP" stringLit
+```
+
+## 注意事项
+
+* `FLASHBACK` 指定的时间点需要在 Garbage Collection (GC) life time 时间内，可以使用系统变量 `tidb_gc_life_time` 配置数据的历史版本的保留时间（默认值是 10m0s）。可以使用以下 SQL 语句查询当前的 safePoint，即 GC 已经清理到的时间点：
+
+```sql
+SELECT * FROM mysql.tidb WHERE variable_name = 'tikv_gc_safe_point';
+```
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-flashback-table-to-timestamp.md:31]:
````suggestion
    ```sql
    SELECT * FROM mysql.tidb WHERE variable_name = 'tikv_gc_safe_point';
    ```
````

File: sql-statements/sql-statement-flashback-table-to-timestamp.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+* 执行 `FLASHBACK TABLE` 需要有 TABLE 权限。
+* 在 `FLASHBACK TABLE` 指定的时间点到开始执行的时间段内不能存在相关的 DDL 执行记录。
+* 在执行 `FLASHBACK TABLE` 期间，TiDB 会主动断开所有相关的链接和正在执行的事务，直到 `FLASHBACK TABLE` 完成。
+* `FLASHBACK TABLE` 命令不能取消，一旦开始执行会一直重试，直到成功。
+
+## 示例
+
+* 恢复新插入的数据
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-flashback-table-to-timestamp.md:40]:
```suggestion
恢复新插入的数据：
```

File: sql-statements/sql-statement-flashback-table-to-timestamp.md [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+|    1 |
++------+
+1 row in set (0.01 sec)
+
+mysql> flashback table t to timestamp '2022-09-28 17:24:16';
+Query OK, 0 rows affected (0.20 sec)
+
+mysql> select * from t;
+Empty set (0.00 sec)
+```
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-flashback-table-to-timestamp.md:73]:
````suggestion
```sql
mysql> CREATE TABLE t(a INT);
Query OK, 0 rows affected (0.09 sec)

mysql> SELECT * FROM t;
Empty set (0.01 sec)

mysql> SELECT NOW();
+---------------------+
| now()               |
+---------------------+
| 2022-09-28 17:24:16 |
+---------------------+
1 row in set (0.02 sec)

mysql> INSERT INTO t VALUES (1);
Query OK, 1 row affected (0.02 sec)

mysql> SELECT * FROM t;
+------+
| a    |
+------+
|    1 |
+------+
1 row in set (0.01 sec)

mysql> FLASHBACK TABLE T TO TIMESTAMP '2022-09-28 17:24:16';
Query OK, 0 rows affected (0.20 sec)

mysql> SELECT * FROM t;
Empty set (0.00 sec)
```
````

File: sql-statements/sql-statement-flashback-table-to-timestamp.md [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+
+## 工作原理
+
+具体工作原理可以参考[`FALSHBACK CLUSTER TO TIMESTAMP`](/sql-statements/sql-statement-flashback-cluster-to-timestamp.md)
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-flashback-table-to-timestamp.md:77]:
```suggestion
具体工作原理可以参考 [`FALSHBACK CLUSTER TO TIMESTAMP`](/sql-statements/sql-statement-flashback-cluster-to-timestamp.md)
```

File: sql-statements/sql-statement-flashback-cluster-to-timestamp.md (hunk 1 above)
Comment (Defined2014) [reply sql-statements/sql-statement-flashback-cluster-to-timestamp.md:104]: add a proposal on TiDB repo, https://github.com/pingcap/tidb/pull/38367/files, so I will remove this part here.

File: sql-statements/sql-statement-flashback-to-timestamp.md [hunk 10]
```diff
@@ -0,0 +1,102 @@
+---
+title: FLASHBACK [CLUSTER | DATABASE | TABLE] TO TIMESTAMP
+---
+
+# FLASHBACK [CLUSTER | DATABASE | TABLE] TO TIMESTAMP
+
+TiDB v6.4.0 引入了 `FLASHBACK [CLUSTER | DATABASE | TABLE] TO TIMESTAMP` 语法，其功能是将集群的数据恢复到特定的时间点。
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-flashback-to-timestamp.md:7]:
```suggestion
TiDB v6.4.0 引入了 `FLASHBACK [CLUSTER | DATABASE | TABLE] TO TIMESTAMP` 语法，其功能是将集群、数据库、数据表的数据恢复到特定的时间点。
```

File: sql-statements/sql-statement-flashback-to-timestamp.md [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+### 语法图
+
+```ebnf+diagram
+FlashbackToTimestampStmt ::=
+    "FLASHBACK" ("TABLE" TableNameList | "DATABASE" DBNameList | "CLUSTER") "TO" "TIMESTAMP" stringLit
+```
+
+## 注意事项
+
+* `FLASHBACK` 指定的时间点需要在 Garbage Collection (GC) life time 时间内，可以使用系统变量 [`tidb_gc_life_time`](/system-variables.md#tidb_gc_life_time-从-v50-版本开始引入) 配置数据的历史版本的保留时间（默认值是 10m0s）。可以使用以下 SQL 语句查询当前的 safePoint，即 GC 已经清理到的时间点：
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-flashback-to-timestamp.md:26]:
```suggestion
* `FLASHBACK` 指定的时间点需要在 Garbage Collection (GC) life time 时间内，可以使用系统变量 [`tidb_gc_life_time`](/system-variables.md#tidb_gc_life_time-从-v50-版本开始引入) 配置数据的历史版本的保留时间（默认值是 `10m0s`）。可以使用以下 SQL 语句查询当前的 safePoint，即 GC 已经清理到的时间点：
```

File: sql-statements/sql-statement-flashback-to-timestamp.md [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+```sql
+SELECT * FROM mysql.tidb WHERE variable_name = 'tikv_gc_safe_point';
+```
+
+* 仅支持拥有对应权限的用户执行该 SQL 命令：
+    * 执行 `FLASHBACK CLUSTER` 操作需要有 `SUPER` 权限。
+    * 执行 `FLASHBACK DATABASE` 操作需要有 `DATABASE` 权限。
+    * 执行 `FLASHBACK TABLE` 操作需要有 `TALBE` 权限。
+* 在 `FLASHBACK` 指定的时间点到开始执行的时间段内不能存在相关表结构变更的 DDL 记录，若存在，TiDB 会拒绝该 DDL 操作。
+* 在执行 `FLASHBACK [CLUSTER | DATABASE | TABLE] TO TIMESTAMP` 前，TiDB 会主动断开所有相关表上的链接，并在禁止对这些表进行读写操作，直到 `FLASHBACK` 完成。
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-flashback-to-timestamp.md:37]:
```suggestion
* 在执行 `FLASHBACK [CLUSTER | DATABASE | TABLE] TO TIMESTAMP` 前，TiDB 会主动断开所有相关表上的连接，并禁止对这些表进行读写操作，直到 `FLASHBACK` 完成。
```

File: sql-statements/sql-statement-flashback-to-timestamp.md [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+* `FLASHBACK [CLUSTER | DATABASE | TABLE] TO TIMESTAMP` 命令不能取消，一旦开始执行 TiDB 会一直重试，直到成功。
+* 用户可以通过日志查看 flashback 执行进度，具体的日志如下所示：
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-flashback-to-timestamp.md:39]:
```suggestion
* 用户可以通过日志查看 `FLASHBACK` 执行进度，具体的日志如下所示：
```

File: sql-statements/sql-statement-flashback-to-timestamp.md [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+# FLASHBACK [CLUSTER | DATABASE | TABLE] TO TIMESTAMP
+
+TiDB v6.4.0 引入了 `FLASHBACK [CLUSTER | DATABASE | TABLE] TO TIMESTAMP` 语法，其功能是将集群的数据恢复到特定的时间点。
+
+## 语法
+
+```sql
+FLASHBACK TABLE tbl1, tbl2 TO TIMESATMP '2022-09-21 16:02:50';
+FLASHBACK DATABASE db1, db2 TO TIMESATMP '2022-09-21 16:02:50';
+FLASHBACK CLUSTER TO TIMESATMP '2022-09-21 16:02:50';
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-flashback-to-timestamp.md:14]:
```suggestion
FLASHBACK CLUSTER TO TIMESATMP '2022-09-21 16:02:50';
FLASHBACK DATABASE db1, db2 TO TIMESATMP '2022-09-21 16:02:50';
FLASHBACK TABLE tbl1, tbl2 TO TIMESATMP '2022-09-21 16:02:50';
```

File: sql-statements/sql-statement-flashback-to-timestamp.md [hunk 15]
```diff
@@ -0,0 +1,102 @@
...
+```
+
+### 语法图
+
+```ebnf+diagram
+FlashbackToTimestampStmt ::=
+    "FLASHBACK" ("TABLE" TableNameList | "DATABASE" DBNameList | "CLUSTER") "TO" "TIMESTAMP" stringLit
+```
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-flashback-to-timestamp.md:22]:
````suggestion
```ebnf+diagram
FlashbackToTimestampStmt ::=
    "FLASHBACK" ("CLUSTER" | "DATABASE" DBNameList | "TABLE" TableNameList) "TO" "TIMESTAMP" stringLit
```
````
image

**Review by bb7133 — APPROVED** [review]
LGTM

**Review by JmPotato — APPROVED** [review]

**Review by Oreoxmt — APPROVED** [review]
LGTM

File: sql-statements/sql-statement-flashback-to-timestamp.md [hunk 16]
```diff
@@ -0,0 +1,103 @@
...
+
+# FLASHBACK [CLUSTER | DATABASE | TABLE] TO TIMESTAMP
+
+TiDB v6.4.0 引入了 `FLASHBACK [CLUSTER | DATABASE | TABLE] TO TIMESTAMP` 语法，其功能是将集群、数据库、数据表的数据恢复到特定的时间点。
+
+## 语法
+
+```sql
+FLASHBACK CLUSTER TO TIMESTAMP '2022-09-21 16:02:50';
+FLASHBACK DATABASE db1, db2 TO TIMESTAMP '2022-09-21 16:02:50';
```
Comment (Defined2014) [inline sql-statements/sql-statement-flashback-to-timestamp.md:14]:
```suggestion
FLASHBACK DATABASE db TO TIMESTAMP '2022-09-21 16:02:50';
```

File: sql-statements/sql-statement-flashback-to-timestamp.md [hunk 17]
```diff
@@ -0,0 +1,103 @@
...
+FLASHBACK TABLE tbl1, tbl2 TO TIMESTAMP '2022-09-21 16:02:50';
+```
+
+### 语法图
+
+```ebnf+diagram
+FlashbackToTimestampStmt ::=
+    "FLASHBACK" ("CLUSTER" | "DATABASE" DBNameList | "TABLE" TableNameList) "TO" "TIMESTAMP" stringLit
```
Comment (Defined2014) [inline sql-statements/sql-statement-flashback-to-timestamp.md:22]:
```suggestion
    "FLASHBACK" ("CLUSTER" | "DATABASE" DBName | "TABLE" TableNameList) "TO" "TIMESTAMP" stringLit
```

File: sql-statements/sql-statement-flashback-to-timestamp.md [hunk 18]
```diff
@@ -0,0 +1,103 @@
...
+FlashbackToTimestampStmt ::=
+    "FLASHBACK" ("CLUSTER" | "DATABASE" DBName | "TABLE" TableNameList) "TO" "TIMESTAMP" stringLit
+```
+
+## 注意事项
+
+* `FLASHBACK` 指定的时间点需要在 Garbage Collection (GC) life time 时间内。你可以使用系统变量 [`tidb_gc_life_time`](/system-variables.md#tidb_gc_life_time-从-v50-版本开始引入) 配置数据的历史版本的保留时间（默认值是 `10m0s`）。可以使用以下 SQL 语句查询当前的 `safePoint`，即 GC 已经清理到的时间点：
+
+    ```sql
+    SELECT * FROM mysql.tidb WHERE variable_name = 'tikv_gc_safe_point';
```
Comment (shichun-0415) [inline sql-statements/sql-statement-flashback-to-timestamp.md:30]: Should we document the variable `tikv_gc_safe_point`?

File: sql-statements/sql-statement-flashback-to-timestamp.md [hunk 19]
```diff
@@ -0,0 +1,103 @@
...
+    ```
+
+* 仅支持拥有对应权限的用户执行该 SQL 命令：
+    * 执行 `FLASHBACK CLUSTER` 操作需要有 `SUPER` 权限。
+    * 执行 `FLASHBACK DATABASE` 操作需要有 `DATABASE` 权限。
+    * 执行 `FLASHBACK TABLE` 操作需要有 `TABLE` 权限。
+* 在 `FLASHBACK` 指定的时间点到开始执行的时间段内不能存在相关表结构变更的 DDL 记录。若存在，TiDB 会拒绝该 DDL 操作。
+* 在执行 `FLASHBACK [CLUSTER | DATABASE | TABLE] TO TIMESTAMP` 前，TiDB 会主动断开所有相关表上的连接，并禁止对这些表进行读写操作，直到 `FLASHBACK` 完成。
+* `FLASHBACK [CLUSTER | DATABASE | TABLE] TO TIMESTAMP` 命令不能取消，一旦开始执行 TiDB 会一直重试，直到成功。
+* 可以通过日志查看 `FLASHBACK` 执行进度，具体的日志如下所示：
```
Comment (shichun-0415) [inline sql-statements/sql-statement-flashback-to-timestamp.md:40]: Line 40-43 不属于注意事项，建议放到示例章节。

File: sql-statements/sql-statement-flashback-to-timestamp.md (hunk 18 above)
Comment (Defined2014) [reply sql-statements/sql-statement-flashback-to-timestamp.md:30]: It's not a system variable, just a line in `mysql.tidb` table.

**Review by shichun-0415 — APPROVED** [review]
LGTM

File: sql-statements/sql-statement-flashback-to-timestamp.md [hunk 20]
```diff
@@ -0,0 +1,104 @@
...
+summary: TiDB 数据库中 FLASHBACK CLUSTER TO TIMESTAMP 的使用概况。
+---
+
+# FLASHBACK CLUSTER TO TIMESTAMP
+
+TiDB v6.4.0 引入了 `FLASHBACK CLUSTER TO TIMESTAMP` 语法，其功能是将集群的数据恢复到特定的时间点。
+
+> **警告：**
+>
+> 在执行 `FLASHBACK CLUSTER TO TIMESTAMP` 之前，需要暂停 PiTR、TiCDC 等工具，待 flashback 执行完成后再启动，不然会造成同步失败等问题。
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-flashback-to-timestamp.md:12]:
```suggestion
> 在执行 `FLASHBACK CLUSTER TO TIMESTAMP` 之前，需要暂停 PITR、TiCDC 等工具，待 `FLASHBACK` 执行完成后再启动，否则会造成同步失败等问题。
```

File: sql-statements/sql-statement-flashback-to-timestamp.md [hunk 21]
```diff
@@ -0,0 +1,104 @@
...
+
+```sql
+FLASHBACK CLUSTER TO TIMESTAMP '2022-09-21 16:02:50';
+```
+
+### 语法图
+
+```ebnf+diagram
+FlashbackToTimestampStmt ::=
+    "FLASHBACK" ("CLUSTER" | "DATABASE" DBName | "TABLE" TableNameList) "TO" "TIMESTAMP" stringLit
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-flashback-to-timestamp.md:24]:
```suggestion
    "FLASHBACK" ("CLUSTER") "TO" "TIMESTAMP" stringLit
```

File: sql-statements/sql-statement-flashback-to-timestamp.md [hunk 22]
```diff
@@ -0,0 +1,104 @@
...
+## 注意事项
+
+* `FLASHBACK` 指定的时间点需要在 Garbage Collection (GC) life time 时间内。你可以使用系统变量 [`tidb_gc_life_time`](/system-variables.md#tidb_gc_life_time-从-v50-版本开始引入) 配置数据的历史版本的保留时间（默认值是 `10m0s`）。可以使用以下 SQL 语句查询当前的 `safePoint`，即 GC 已经清理到的时间点：
+
+    ```sql
+    SELECT * FROM mysql.tidb WHERE variable_name = 'tikv_gc_safe_point';
+    ```
+
+* 仅支持拥有对应权限的用户执行该 SQL 命令：
+    * 执行 `FLASHBACK CLUSTER` 操作需要有 `SUPER` 权限。
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-flashback-to-timestamp.md:36]:
```suggestion
* 执行 `FLASHBACK CLUSTER` 操作的用户需要有 `SUPER` 权限。
```

File: sql-statements/sql-statement-flashback-to-timestamp.md (hunk 21 above)
Comment (Defined2014) [reply sql-statements/sql-statement-flashback-to-timestamp.md:24]: 这个语法的部分已经加了，需要删掉么

File: sql-statements/sql-statement-flashback-to-timestamp.md [hunk 23]
```diff
@@ -0,0 +1,103 @@
...
+summary: TiDB 数据库中 FLASHBACK CLUSTER TO TIMESTAMP 的使用概况。
+---
+
+# FLASHBACK CLUSTER TO TIMESTAMP
+
+TiDB v6.4.0 引入了 `FLASHBACK CLUSTER TO TIMESTAMP` 语法，其功能是将集群的数据恢复到特定的时间点。
+
+> **警告：**
+>
+> 在执行 `FLASHBACK CLUSTER TO TIMESTAMP` 之前，需要暂停 PITR、TiCDC 等工具，待 `FLASHBACK` 执行完成后再启动，否则会造成同步失败等问题。
```
Comment (shichun-0415) [inline sql-statements/sql-statement-flashback-to-timestamp.md:12]:
```suggestion
> 在执行 `FLASHBACK CLUSTER TO TIMESTAMP` 之前，需要暂停 PITR 和 TiCDC 等工具上运行的同步任务，待 `FLASHBACK` 执行完成后再启动，否则会造成同步失败等问题。
```

**Review by shichun-0415 — APPROVED** [review]
Rest LGTM

**Review by Oreoxmt — APPROVED** [review]
LGTM

File: sql-statements/sql-statement-flashback-to-timestamp.md [hunk 24]
```diff
@@ -0,0 +1,103 @@
...
+summary: TiDB 数据库中 FLASHBACK CLUSTER TO TIMESTAMP 的使用概况。
+---
+
+# FLASHBACK CLUSTER TO TIMESTAMP
+
+TiDB v6.4.0 引入了 `FLASHBACK CLUSTER TO TIMESTAMP` 语法，其功能是将集群的数据恢复到特定的时间点。
+
+> **警告：**
+>
+> 在执行 `FLASHBACK CLUSTER TO TIMESTAMP` 之前，需要暂停 PITR 和 TiCDC 等工具上运行的同步任务，待 `FLASHBACK` 执行完成后再启动，否则会造成同步失败等问题。
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-flashback-to-timestamp.md:12]:
```suggestion
> **警告：**
>
> - 当前该功能为实验特性，不建议在生产环境中使用。
> - 在执行 `FLASHBACK CLUSTER TO TIMESTAMP` 之前，需要暂停 PITR 和 TiCDC 等工具上运行的同步任务，待 `FLASHBACK` 执行完成后再启动，否则会造成同步失败等问题。
```

File: experimental-features.md [hunk 25]
```diff
@@ -36,6 +36,7 @@ aliases: ['/docs-cn/dev/experimental-features-4.0/','/zh/tidb/dev/experimental-f
 + [自定义变量](/user-defined-variables.md#用户自定义变量)。（v2.1 实验特性）
 + [Cascades Planner](/system-variables.md#tidb_enable_cascades_planner)：基于 Cascades 框架的自顶向下查询优化器。（v3.0 实验特性）
 + [`ALTER TABLE ... COMPACT`](/sql-statements/sql-statement-alter-table-compact.md)。（v6.1.0 实验特性）
++ [将集群的数据恢复到特定的时间点 `FLASHBACK CLUSTER TO TIMESTAMP`](/sql-statements/sql-statement-flashback-to-timestamp.md) (v6.4.0 实验特性)
```
Comment (Oreoxmt) [inline experimental-features.md:39]:
```suggestion
+ [将集群的数据恢复到特定的时间点 `FLASHBACK CLUSTER TO TIMESTAMP`](/sql-statements/sql-statement-flashback-to-timestamp.md)。(v6.4.0 实验特性)
```

File: experimental-features.md [hunk 26]
```diff
@@ -36,6 +36,7 @@ aliases: ['/docs-cn/dev/experimental-features-4.0/','/zh/tidb/dev/experimental-f
 + [自定义变量](/user-defined-variables.md#用户自定义变量)。（v2.1 实验特性）
 + [Cascades Planner](/system-variables.md#tidb_enable_cascades_planner)：基于 Cascades 框架的自顶向下查询优化器。（v3.0 实验特性）
 + [`ALTER TABLE ... COMPACT`](/sql-statements/sql-statement-alter-table-compact.md)。（v6.1.0 实验特性）
++ [将集群的数据恢复到特定的时间点 `FLASHBACK CLUSTER TO TIMESTAMP`](/sql-statements/sql-statement-flashback-to-timestamp.md)。(v6.4.0 实验特性)
```
Comment (Oreoxmt) [inline experimental-features.md:39]:
```suggestion
+ [将集群的数据恢复到特定的时间点 `FLASHBACK CLUSTER TO TIMESTAMP`](/sql-statements/sql-statement-flashback-to-timestamp.md)。（v6.4.0 实验特性）
```

File: experimental-features.md [hunk 27]
```diff
@@ -34,11 +34,14 @@ aliases: ['/docs-cn/dev/experimental-features-4.0/','/zh/tidb/dev/experimental-f
 + [生成列](/generated-columns.md#生成列)。（v2.1 实验特性）
 + [自定义变量](/user-defined-variables.md#用户自定义变量)。（v2.1 实验特性）
 + [Cascades Planner](/system-variables.md#tidb_enable_cascades_planner)：基于 Cascades 框架的自顶向下查询优化器。（v3.0 实验特性）
++ [`ALTER TABLE ... COMPACT`](/sql-statements/sql-statement-alter-table-compact.md)。（v6.1.0 实验特性）
```
Comment (Oreoxmt) [inline experimental-features.md:37]:
```suggestion
```

File: experimental-features.md [hunk 28]
```diff
@@ -34,11 +34,13 @@ aliases: ['/docs-cn/dev/experimental-features-4.0/','/zh/tidb/dev/experimental-f
 + [生成列](/generated-columns.md#生成列)。（v2.1 实验特性）
 + [自定义变量](/user-defined-variables.md#用户自定义变量)。（v2.1 实验特性）
 + [Cascades Planner](/system-variables.md#tidb_enable_cascades_planner)：基于 Cascades 框架的自顶向下查询优化器。（v3.0 实验特性）
++ [将集群的数据恢复到特定的时间点 `FLASHBACK CLUSTER TO TIMESTAMP`](/sql-statements/sql-statement-flashback-to-timestamp.md)。（v6.4.0 实验特性）
```
Comment (Oreoxmt) [inline experimental-features.md:37]:
```suggestion
```

File: experimental-features.md [hunk 29]
```diff
@@ -34,11 +34,13 @@ aliases: ['/docs-cn/dev/experimental-features-4.0/','/zh/tidb/dev/experimental-f
...
 + [表级锁 (Table Lock)](/tidb-configuration-file.md#enable-table-lock-从-v400-版本开始引入)（v4.0.0 实验特性）
 + [元数据锁](/metadata-lock.md)（v6.3.0 实验特性）
 + [Range INTERVAL 分区](/partitioned-table.md#range-interval-分区)（v6.3.0 实验特性）
 + [添加索引加速](/system-variables.md#tidb_ddl_enable_fast_reorg-从-v630-版本开始引入)（v6.3.0 实验特性）
```
Comment (Oreoxmt) [inline experimental-features.md:42]:
```suggestion
+ [将集群的数据恢复到特定的时间点 `FLASHBACK CLUSTER TO TIMESTAMP`](/sql-statements/sql-statement-flashback-to-timestamp.md)。（v6.4.0 实验特性）
```

**Review by shichun-0415 — APPROVED** [review]
LGTM

**Pull request merged**